Smt/Query.lean

```diff
@@ -93,22 +94,35 @@ def natConstAssert (n : String) (args : List Name) : Term → MetaM Term
       | [] => Symbol n
       | t :: ts => App (applyList n ts) (Symbol t.toString)
 
--- TODO: support recursive functions.
+-- TODO: support mutually recursive functions.
 partial def toDefineFun (s : Solver) (e : Expr) : MetaM Solver := do
-  let defn : Expr ← Meta.unfoldDefinition e
-  pure (defineFun s id (← params defn) (← type (← Meta.inferType defn)) (← body defn))
+  let some eqnThm ← getUnfoldEqnFor? (nonRec := true) e.constName!
+    | throwError "failed to retrieve equation theorem"
+  let eqnInfo ← getConstInfo eqnThm
+  let d := eqnInfo.type
+  let mutRecFuns := ConstantInfo.all (← getConstInfo e.constName!)
+  trace[smt.debug.query] "mutually recursive functions with {e}: {mutRecFuns}"
+  if mutRecFuns.length > 1 then
+    throwError "mutually recursive functions are not yet supported"
+  if Util.countConst d e.constName! > 1 then
```

Comment:
```suggestion
  -- TODO: Replace by `DefinitionVal.isRec` check when it gets added to Lean core
  if Util.countConst d e.constName! > 1 then
```
We want to change this when Leo adds the API, right?

Reply:
I think he thought that we only wanted `ConstantInfo.all`, which he did add...

Reply:
Oh maybe I misunderstood, I thought there would be further additions to distinguish recursive/non-recursive functions which is not possible with `ConstantInfo.all`. I guess let's leave the comment and see.